Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -604,14 +575,60 @@ function New-LDAPSIdentitySource {
 
     $Password = $Credential.GetNetworkCredential().Password
     $DestinationFileArray = @()
+    $remoteComputers = , $PrimaryUrl
+    if ($PSBoundParameters.ContainsKey('SecondaryUrl')) {
+        $remoteComputers += $SecondaryUrl
+    }
+
+    # check the connection between domain servers and the vcenter
+    foreach ($computerUrl in $remoteComputers) {
+        if (![uri]::IsWellFormedUriString($computerUrl, 'Absolute')) {
+            throw "Incorrect Url format entered from: $computerUrl"
+        }
+        $ParsedUrl = [System.Uri]$computerUrl
+        if ($ParsedUrl.Port -lt 0 -OR $ParsedUrl.Host -eq "" -OR $ParsedUrl.Scheme -eq "") {
+            throw "Incorrect Url format entered from: $computerUrl. The correct Url format is protocol://host:port (Example: ldaps://yourserver.com:636)."
+        }
+        $ResultUrlString = $ParsedUrl.GetLeftPart([UriPartial]::Authority)
+        $ResultUrl = [System.Uri]$ResultUrlString
+        if ($ResultUrl.Host -match "^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$" -and [bool]($ResultUrl.Host -as [ipaddress])) {
+            throw "Incorrect Url format. $computerUrl is an IP address. Please use the hostname exactly as specified on the issued certificate."
+        }
+        # dns lookup
+        try {
+            $Command = 'nslookup ' + $ResultUrl.Host + ' -type=soa'
+            $SSHRes = Invoke-SSHCommand -Command $Command -SSHSession $SSH_Sessions['VC'].Value
+        }
+        catch {
+            throw "The FQDN $($ResultUrl.Host) cannot be resolved to an IP address. Make sure DNS is configured."
+        }
+        Write-Host "The FQDN $($ResultUrl.Host) is resolved successfully."
+        # reverse dns lookup
+        try {
+            $IPAddress = $SSHRes.Output | grep "Address:" | grep -v "#" | awk '{print $2}' | head -n 1
```

Comment:
Let's parse this using PS facilities, I don't know if we install grep/awk and we shouldn't expect the arguments/behavior to remain compatible if we have to use different OS base image.